### ISSUE TITLE
loss_per_head seems wrong

### ISSUE THREAD

**Issue by haoosz** [DonkeyShot21/UNO#8]
Hi, I think the calculation of `loss_per_head` is wrong.
```
def cross_entropy_loss(self, preds, targets): # n_heads, batch_size, logits
        preds = F.log_softmax(preds / self.hparams.temperature, dim=-1)
        return -torch.mean(torch.sum(targets * preds, dim=-1))
```
this code averages both 0 and 1 dims so the loss is not head-wise. When conducting `self.loss_per_head += loss_cluster.clone().detach()`, each head loss adds the same mean value.
```
 return -torch.mean(torch.sum(targets * preds, dim=-1), dim=-1) # revised
```
I think this would be good. If there's anything wrong with my understanding, please correct me. Thanks.

**Comment by DonkeyShot21** [comment]
Yes, you are right. We made some changes to the repository recently and forgot to update this. I will fix it soon.

**Comment by haoosz** [comment]
Thanks for the reply.
